### ISSUE TITLE
Add spesific version of library

### ISSUE THREAD

**Issue by azzamsa** [Pelican-Elegant/documentation#62]
I think it's better to add specific version explicitly to library in requirements.txt and requirements-test.txt. So we know every changes made by upstream.

Also we can add `rev` in peru using [peru reup](https://github.com/buildinspace/peru#magical-updates)

We don't want to spend lot times digging problem here, but turn out it comes from the changes made in upstream. e.g https://github.com/Pelican-Elegant/elegant/issues/254

**Comment by iranzo** [comment]
Elegant is not distributing requirements nor peru yaml, as it's just a theme, and each of the extra components that can be used should be optional.

Another thing is to work on a canned setup that can be automated  as suggested in #63

**Comment by azzamsa** [comment]
> Elegant is not distributing requirements nor peru yaml, as it's just a theme, and each of the extra components that can be used should be optiona

I see, but how awesome if one can fork this documentation site then just need small changes here and there.

About your article, great addition for this docs.

**Comment by iranzo** [comment]
@azzamsa check #63

**Comment by azzamsa** [comment]
@talha131 Thanks a lot for your explanation. My 'own' problem with elegant is:

When I set the theme to elegant, I can't get tipue_search working, spend huge time with the help from Iranzo, turns out the problem was from elegant search template. It still use json version instead of js. When I run peru sync, indeed peru will grab the latest version. That's why I highly recommend for requirements or peru.yml to have `rev` or `explicit version`. So elegant code still relevant even if new user grab its pelicanconf.py and its peru.yml and use in their own machine. Cause peru sync will grab the spesific `rev` not the latest version.

That's all what I mean.

I tried mimicking publishconf from this docs, from your repo, nothing works until two days, then I find out that the upstream version of tipue was changed. If only we specified the version of plugins or its `rev` in peru.yml, this will never happen.

Hope you consider this.

**Comment by talha131** [comment]
Thank you @azzamsa for the detailed and thorough explanation. 

The reason you had to dealt with tipue search issue is because we have messed up releasing 2.0. The search features relies on the plugin and tipue code, both of which upgraded recently, and we failed to update the theme timely. Once we are out of this trough, you will see more stable and quicker releases from us.

`peru` should not be used to resolve search issue. `peru` is mainly added here by @iranzo to facilitate the CI. We can't and should not expect our users to dabble with `peru` at all, because it is not a requirement or dependency of either Elegant or Pelican.

Lastly, we want `peru.yml` to always fetch the latest plugins so that if something is broken it is brought to our attention sooner. 

So bottom line is you faced the broken search issue because our release is not ready yet. Pinning version in `peru.yml` is a temporary work around.

Finally lets wait for #63. I am very hopeful it will solve the problem you mentioned, for now and all future releases.

**Comment by azzamsa** [comment]
Oh I see, thank you for complete explanation. Sure, I saw lot of 'best theme for pelican' query in search engines point to elegant.  People also recommend it in forums. Hope elegant get more stable relases

Indeed, I didn't know peru before. I just know it here. I agree with your opinion about peru. If we depend a lot to it for this issue, it just add more barriers for new comers.

> Finally lets wait for #63. I am very hopeful it will solve the problem you mentioned, for now and all future releases.

:) 

Thanks a lot Talha.
